File: solutions/Virtual-Assistant/src/csharp/microsoft.bot.solutions/Models/SkillEvent.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Collections.Generic;
using Microsoft.Bot.Configuration;
using Newtonsoft.Json;

namespace Microsoft.Bot.Solutions.Model
{
    public class SkillEvent : ConnectedService
    {
        public SkillEvent()
            : base("skillEvent")
        {
        }

        [JsonProperty("event")]
        public string Event { get; set; }

        [JsonProperty("skillIds")]
        public string[] SkillIds { get; set; }

        [JsonProperty("parameters")]
        public Dictionary<string, string> Parameters { get; set; }
    }
}

File: solutions/Virtual-Assistant/src/typescript/botbuilder-solutions/src/skills/skillEvent.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { ConnectedService, ServiceTypes } from 'botframework-config';

export class SkillEvent extends ConnectedService {

    public event: string = '';

    public skillIds: string[] = [];

    public parameters: Map<string, string> = new Map();

    constructor() {
        super(undefined, ServiceTypes.Generic);
    }
}
